### PR TITLE
Client: support execute and prepare 

### LINKED ISSUE #34: Add execute statement
There is not execute in client.
So it can't be used for SQL expressions.

I've tried to add this functionality, but with no success. It works too bad - hangs up on getting queries results (but queries are done - tables created and dropped in fact, for example).

```
// I guess that error is in this method. But I don't know, what I did wrong
template<class BUFFER>
template <class T>
size_t
RequestEncoder<BUFFER>::encodeExecute(const std::string &expr, const T &args)
{
	iterator_t<BUFFER> request_start = m_Buf.end();
	m_Buf.write('\xce');
	m_Buf.write(uint32_t{0});
	encodeHeader(Iproto::EXECUTE);
	tnt::mpp::encode(m_Buf, tnt::mpp::as_map(std::forward_as_tuple(
		MPP_AS_CONST(Iproto::SQL_TEXT), expr,
		MPP_AS_CONST(Iproto::SQL_BIND), mpp::as_arr(args))));
	uint32_t request_size = (m_Buf.end() - request_start) - PREHEADER_SIZE;
	++request_start;
	request_start.set(__builtin_bswap32(request_size));
	return request_size + PREHEADER_SIZE;
}

template<class BUFFER, class NetProvider>
template <class T>
rid_t
Connection<BUFFER, NetProvider>::execute(const std::string &expr, const T &args)
{
	impl->enc.encodeExecute(expr, args);
	impl->connector.readyToSend(*this);
	return RequestEncoder<BUFFER>::getSync();
}
```

### PR DESCRIPTION
**Pull request by drewdzzz**
The patch provides client with execute and prepare requests. Execute allows to pass an SQL statement or id of a prepared statement and parameters for it. Response body is populated with sql fields.

Closes #34

### CODE REVIEW

File: src/Client/Connection.hpp [hunk 1]
```diff
@@ -177,6 +177,34 @@ class Connection
...
 	rid_t call(const std::string &func, const T &args);
 	rid_t ping();
+	
+	/**
+	 * Execute the SQL statement contained in the 'statement' parameter.
+	 * @param statement statement, which should conform to the rules for SQL grammar
+	 * @param parameters tuple for placeholders in the statement
+	 * @retval request id
+	 */
+
```
Comment (CuriousGeorgiy) [inline src/Client/Connection.hpp:187]: Nit: excessive newline.

**Review by CuriousGeorgiy — CHANGES_REQUESTED** [review]
Overall, looks great, and very clean!

There was an explicit request that we promised to fulfill: please add some examples to the `./examples`. Maybe a separate example or as part of the `Simple` one.

File: src/Client/Connection.hpp (hunk 1 above)
Comment (drewdzzz) [reply src/Client/Connection.hpp]: Fixed.

**Review by alyapunov — APPROVED** [review]

**Comment by drewdzzz** [comment]
@CuriousGeorgiy Added a new example. Please check if it clearly explains all capabilities of SQL in tntcxx.

**Review by CuriousGeorgiy — APPROVED** [review]

**Pull request merged**